src/wp-includes/class-wp-comment-query.php

```diff
@@ -770,6 +770,11 @@ protected function get_comment_ids() {
 			'NOT IN' => (array) $this->query_vars['type__not_in'],
 		);
 
+		// @since 6.9.0 Exclude the 'note' comment type, unless 'all' types or the 'note' type explicitly are requested.
+		if ( ! in_array( 'all', $raw_types['IN'], true ) && ! in_array( 'note', $raw_types['IN'], true ) ) {
+			$raw_types['NOT IN'][] = 'note';
```

Comment:
Do we check that it is not already in the `NOT IN` array?

Reply:
👍 I will add that, I noticed that check was missing as well.

Reply:
Added in f8ca7e67b7c454bbf0c0e31daa8489bcd2af12de